cryptofeed/exchange/okcoin.py

```diff
@@ -204,7 +240,34 @@ async def message_handler(self, msg: str, conn, timestamp: float):
                 await self._book(msg, timestamp)
             elif 'swap/funding_rate' in msg['table']:
                 await self._funding(msg, timestamp)
+            elif 'spot/order' in msg['table']:
+                await self._order(msg, timestamp)
             else:
                 LOG.warning("%s: Unhandled message %s", self.id, msg)
         else:
             LOG.warning("%s: Unhandled message %s", self.id, msg)
+    
+    def connect(self) -> List[Tuple[AsyncConnection, Callable[[None], None], Callable[[str, float], None]]]:
+        ret = []
+        for channel in self.subscription or self.channels:
+            if is_authenticated_channel(channel):
+                if channel in self.subscription:
+                    ret.append((AsyncConnection(self.address, self.id, **self.ws_defaults), partial(self.user_order_subscribe, symbol=self.subscription.get(channel).pop()), self.message_handler))
```

Comment:
it looks like authenticated subscribe only takes 1 symbol. If you pop off the contents of the sub dict you could have a list of symbols. If you pop it off, on reconnects, it will be lost. I think you can make it something like this:

```python
if is_authenticated_channel(channel):
    syms = self.symbols or self.subscription[channel]
   for s in syms:
       ret.append((AsyncConnection(self.address, self.id, **self.ws_defaults), partial(self.user_order_subscribe, symbol=s), self.message_handler))
```

Thats obviously untested, so if you use that test it first

Reply:
incorporated - thanks